providers/apis/dydx/multi_market_map_fetcher.go

```diff
@@ -142,29 +158,71 @@ func (f *MultiMarketMapRestAPIFetcher) Fetch(ctx context.Context, chains []mmcli
 	wg.Wait()
 
 	dydxMainnetMarketMapResponse := <-dydxMainnetResponseChan
-
 	dydxResearchMarketMapResponse := <-dydxResearchResponseChan
 
-	// combine the two market maps
 	// if the dydx mainnet market-map response failed, return the dydx mainnet failed response
 	if _, ok := dydxMainnetMarketMapResponse.UnResolved[DYDXChain]; ok {
 		f.logger.Error("dydx mainnet market-map fetch failed", zap.Any("response", dydxMainnetMarketMapResponse))
 		return dydxMainnetMarketMapResponse
 	}
 
+	// if the dydx research market-map response failed, return the dydx research failed response
+	if _, ok := dydxResearchMarketMapResponse.UnResolved[DYDXChain]; ok {
+		f.logger.Error("dydx research market-map fetch failed", zap.Any("response", dydxResearchMarketMapResponse))
+		return dydxResearchMarketMapResponse
+	}
+
 	// otherwise, add all markets from dydx research
 	dydxMainnetMarketMap := dydxMainnetMarketMapResponse.Resolved[DYDXChain].Value.MarketMap
 
-	if resolved, ok := dydxResearchMarketMapResponse.Resolved[DYDXChain]; ok {
+	resolved, ok := dydxResearchMarketMapResponse.Resolved[DYDXChain]
+	if ok {
 		for ticker, market := range resolved.Value.MarketMap.Markets {
 			// if the market is not already in the dydx mainnet market-map, add it
 			if _, ok := dydxMainnetMarketMap.Markets[ticker]; !ok {
 				f.logger.Debug("adding market from dydx research", zap.String("ticker", ticker))
 				dydxMainnetMarketMap.Markets[ticker] = market
 			}
 		}
-	} else {
-		return dydxResearchMarketMapResponse
+	}
+
+	// if the fetcher is only for CoinMarketCap markets, filter out all non-CMC markets
+	if f.isCMCOnly {
+		for ticker, market := range dydxMainnetMarketMap.Markets {
+			market.Ticker.MinProviderCount = 1
+			dydxMainnetMarketMap.Markets[ticker] = market
+
+			var (
+				seenCMC     = false
+				cmcProvider mmtypes.ProviderConfig
+			)
+
+			for _, provider := range market.ProviderConfigs {
+				if provider.Name == coinmarketcap.Name {
+					seenCMC = true
+					cmcProvider = provider
+				}
+			}
+
+			// if we saw a CMC provider, add it to the market
+			if seenCMC {
+				market.ProviderConfigs = []mmtypes.ProviderConfig{cmcProvider}
+				dydxMainnetMarketMap.Markets[ticker] = market
+				continue
+			}
+
+			// If we did not see a CMC provider, we can attempt to add it using the CMC marketmap
+			cmcMarket, ok := marketmaps.CoinMarketCapMarketMap.Markets[ticker]
+			if !ok {
+				f.logger.Info("did not find CMC market for ticker", zap.String("ticker", ticker))
+				delete(dydxMainnetMarketMap.Markets, ticker)
```

Comment:
Does this mean we can create invalid market-maps? I.e we remove an adjust by

Reply:
technically yes, but all of the important adjust bys are in the constants dir. this is a temp fix till we fully move to marketmap generation.